### PR TITLE
TS-29789 Include/excludes not considered correctly.

### PR DESCRIPTION
**Pull request by stahlbauer**
Addresses issue [TS-29789](https://cqse.atlassian.net/browse/TS-29789)

- [x] Changes are tested adequately
- [x] CHANGELOG.md of the affected packages are updated

Please respect the vote of the [Teamscale bot](https://demo.teamscale.com) or flag irrelevant findings as tolerated or false positives. If you feel that the Teamscale config needs adjustment, please state so in a comment and discuss this with your reviewer.

### CODE REVIEW

File: packages/teamscale-javascript-instrumenter/src/instrumenter/Instrumenter.ts [hunk 1]
```diff
@@ -146,19 +145,27 @@ export class IstanbulInstrumenter implements IInstrumenter {
...
 					sourcePattern
-				);
-
-				// The process also can result in a new source map that we will append in the result.
-				//
-				// `lastSourceMap` === Sourcemap for the last file that was instrumented.
-				finalSourceMap = convertSourceMap.fromObject(instrumenter.lastSourceMap()).toComment();
-
-				break;
+				).then(instrumentedSource => {
```
Comment (DreierF) [inline packages/teamscale-javascript-instrumenter/src/instrumenter/Instrumenter.ts:153]: Why was this moved into a `then`? As the method is already `async` using `await` instead would be more readable

File: packages/teamscale-javascript-instrumenter/src/instrumenter/Instrumenter.ts [hunk 2]
```diff
@@ -193,25 +191,30 @@ export class IstanbulInstrumenter implements IInstrumenter {
...
-		const instrumentedSourceMapConsumer: SourceMapConsumer | undefined = await this.loadSourceMap(
+		return await this.loadSourceMap(
 			instrumentedSource, taskElement.fromFile
-		);
-
-		// Without a source map, excludes/includes do not work.
-		if (!instrumentedSourceMapConsumer) {
-			return instrumentedSource;
-		}
+		).then((instrumentedSourceMapConsumer: SourceMapConsumer | undefined) => {
```
Comment (DreierF) [inline packages/teamscale-javascript-instrumenter/src/instrumenter/Instrumenter.ts:196]: Same here

**Review by DreierF — CHANGES_REQUESTED** [review]

File: packages/teamscale-javascript-instrumenter/src/instrumenter/Instrumenter.ts (hunk 2 above)
Comment (stahlbauer) [reply packages/teamscale-javascript-instrumenter/src/instrumenter/Instrumenter.ts]: I have reverted this to async/await.

File: packages/teamscale-javascript-instrumenter/src/instrumenter/Instrumenter.ts (hunk 1 above)
Comment (stahlbauer) [reply packages/teamscale-javascript-instrumenter/src/instrumenter/Instrumenter.ts]: I have reverted this to async/await.

**Pull request merged**